File: shared/src/main/java/ru/privatenull/pnauth/api/AuthApi.java
package ru.privatenull.pnauth.api;

import java.util.Optional;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;
import ru.privatenull.pnauth.kernel.ExtensionKernel;

/** Public service exposed by both proxy adapters. */
public interface AuthApi extends AuthSessionApi, AuthTotpApi, AuthAdminApi, AuthAdmissionApi, AuthDialogApi,
        AuthEventApi, AuthExtensionApi, AuthDisplayApi, ExtensionKernel, AutoCloseable {
    Optional<AuthUser> user(UUID uniqueId);

    CompletableFuture<AuthResult> togglePremium(UUID uniqueId);

    CompletableFuture<AuthResult> unregister(UUID uniqueId, String password);

    @Override
    void close();
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/api/AuthApi.kt
package ru.privatenull.pnauth.api

import ru.privatenull.pnauth.kernel.ExtensionKernel
import java.util.Optional
import java.util.UUID
import java.util.concurrent.CompletableFuture

/** Public service exposed by both proxy adapters. */
interface AuthApi : AuthSessionApi, AuthTotpApi, AuthAdminApi, AuthAdmissionApi, AuthDialogApi,
    AuthEventApi, AuthExtensionApi, AuthDisplayApi, ExtensionKernel, AutoCloseable {
    fun user(uniqueId: UUID): Optional<AuthUser>

    fun togglePremium(uniqueId: UUID): CompletableFuture<AuthResult>

    fun unregister(uniqueId: UUID, password: String): CompletableFuture<AuthResult>

    override fun close()
}
